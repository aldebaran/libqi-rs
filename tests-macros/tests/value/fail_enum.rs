#[derive(qi_macros::Reflect, qi_macros::ToValue, qi_macros::IntoValue, qi_macros::FromValue)]
#[qi(value = "qi_value", transparent)]
enum E {
    A,
    B(i32),
    C { c: String },
}

fn main() {}
