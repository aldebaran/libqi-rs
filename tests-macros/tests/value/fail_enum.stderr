error: Reflect cannot be derived on enums and unions
 --> tests/value/fail_enum.rs:3:6
  |
3 | enum E {
  |      ^

error: ToValue cannot be derived on enums and unions
 --> tests/value/fail_enum.rs:3:6
  |
3 | enum E {
  |      ^

error: IntoValue cannot be derived on enums and unions
 --> tests/value/fail_enum.rs:3:6
  |
3 | enum E {
  |      ^

error: FromValue cannot be derived on enums and unions
 --> tests/value/fail_enum.rs:3:6
  |
3 | enum E {
  |      ^
